django_unicorn/utils.py

```diff
@@ -72,33 +72,49 @@ def dicts_equal(dictionary_one: Dict, dictionary_two: Dict) -> bool:
     return is_valid
 
 
-def get_cacheable_component(
-    component: "django_unicorn.views.UnicornView",
-) -> "django_unicorn.views.UnicornView":
+class CacheableComponent:
```

Comment:
I'll be interested in what you think of this approach. I found having to create the cacheable component, then restore back the extra_context/request everywhere was scary. Meaning I could forget a spot and not notice. I would hope the enter/exit/with approach would help with that, but maybe its awkward.

Reply:
This is a much better pattern and using a context manager totally makes sense for this use case.